src/tunnel.cc

```diff
@@ -393,16 +397,15 @@ TunnelStateData::TunnelStateData(ClientHttpRequest *clientRequest) :
     committedToServer(false),
     n_tries(0),
     banRetries(nullptr),
-    codeContext(CodeContext::Current())
+    codeContext(CodeContext::Current()),
+    peeringTimer(((assert(clientRequest && clientRequest->request)), clientRequest->request->hier.totalResponseTime()))
```

Comment:
We need to find a way how to guarantee clientRequest and request here and remove this clumsy construction.